x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/wired_stream.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  StreamDefinition,
+  WiredStreamDefinition,
+  isWiredStreamDefinition,
+} from '@kbn/streams-schema';
+import { IScopedClusterClient } from '@kbn/core/server';
+import { Logger } from '@kbn/logging';
+import { InternalIStorageClient, ApplicationDocument } from '@kbn/storage-adapter';
+import { cloneDeep } from 'lodash';
+import { isResponseError } from '@kbn/es-errors';
+import { State } from './state';
+import { StreamActiveRecord, ValidationResult } from './stream_active_record';
+import { syncWiredStreamDefinitionObjects } from '../helpers/sync';
+import { deleteStreamObjects } from '../stream_crud';
+
+export class WiredStream extends StreamActiveRecord<WiredStreamDefinition> {
+  constructor(definition: WiredStreamDefinition) {
+    super(definition); // Maybe the subclass should double check here that it was really given the right type of definition?
+    // What about the assets?
+  }
+
+  public clone(): StreamActiveRecord<WiredStreamDefinition> {
```

Comment:
@simianhacker I think clone has to be abstract since it has to return the subclass instance?

Reply:
Agreed